source/minio-features/erasure-coding.rst

```diff
@@ -0,0 +1,277 @@
+- ``EC:N`` Parity Blocks.
+
+MinIO randomly and uniformly distributes the data and parity blocks across
+drives in the erasure set with *no overlap*. While a drive may contain both data
+and parity blocks for multiple unique objects, a single unique object has no
+more than one block per drive in the set. For versioned objects, MinIO selects
+the same drives for both data and parity storage while maintaining zero overlap
+on any single drive.
+
+The specified parity for an object also dictates the minimum number of Erasure
+Set drives ("Quorum") required for MinIO to either read or write that object:
+
+Read Quorum
+   The minimum number of Erasure Set drives required for MinIO to 
+   serve read operations. MinIO can automatically reconstruct an object
+   with corrupted or missing data blocks if enough drives are online to
+   provide Read Quorum for that object.
+  
+   MinIO Read Quorum is ``DRIVES - (EC:N)``.
+
+Write Quorum
+  The minimum number of Erasure Set drives required for MinIO
+  to serve write operations. MinIO requires enough available drives to
+  eliminate the risk of split-brain scenarios. 
+  
+  MinIO Write Quorum is ``DRIVES - (EC:N-1)``.
+
+Storage Classes
+~~~~~~~~~~~~~~~
+
+MinIO supports storage classes with Erasure Coding to allow applications to
+specify per-object :ref:`parity <minio-ec-parity>`. Each storage class specifies
+a ``EC:N`` parity setting to apply to objects created with that class. 
+
+MinIO storage classes are *distinct* from Amazon Web Services storage classes.
+MinIO storage classes define *parity settings per object*, while AWS storage
+classes define *storage tiers per object*. 
+
+MinIO provides the following two storage classes:
+
+``STANDARD``
+   The ``STANDARD`` storage class is the default class for all objects. 
+
+   You can configure the ``STANDARD`` storage class parity using either:
+
+   - The :envvar:`MINIO_STORAGE_CLASS_STANDARD` environment variable, *or*
+   - The :mc:`mc admin config` command to modify the ``storage_class.standard``
+     configuration setting.
+
+   Starting with <RELEASE>, MinIO defaults ``STANDARD`` storage class to
+   ``EC:4``.
+
+   The maximum possible parity value is half of the total drives in the
+   :ref:`Erasure Set <minio-ec-erasure-set>`.
+
+   ``STANDARD`` parity *must* be greater than ``REDUCED_REDUNDANCY``. If
+   ``REDUCED_REDUNDANCY`` is unset, ``STANDARD`` parity *must* be greater
+   than 2
+
+``REDUCED_REDUNDANCY``
+   The ``REDUCED_REDUNDANCY`` storage class allows creating objects with
+   lower parity than ``STANDARD``. 
+
+   You can configure the ``REDUCED_REDUNDANCY`` storage class parity using
+   either:
+
+   - The :envvar:`MINIO_STORAGE_CLASS_REDUCED` environment variable, *oir*
+   - The :mc:`mc admin config` command to modify the 
+     ``storage_class.rrs`` configuration setting.
+
+   The default value is ``EC:2``.
+
+   ``REDUCED_REDUNDANCY`` parity *must* be less than ``STANDARD``. If 
```

Comment:
Technical nit, current release will allow you to set RRS equal to STANDARD, but there is no point in doing that since it would take away any usefulness, not sure if it is worth calling out or not

Reply:
That kind of depends on what we want to do from an engineering perspective. We can imply that it should always be less than (because equal to makes no sense). Or we can be strictly accurate here and let users decide if they want RR to be = to STANDARD

Reply:
Yes, was referencing the pedantic use case just in case we cared to call it out.  All in all, RRS is really not used very much so I don't know that we need to cover the topic in any more depth than we already have here.

Reply:
Hm. In that case I'll just change to "less than or equal to" and let users decide whether they would ever want to do that. I suppose an admin might want to do that.